packages/components/dialog/src/use-dialog.ts

```diff
@@ -155,14 +154,21 @@ export const useDialog = (
     }
   }
 
+  watch(
+    () => props.zIndex,
+    () => {
+      zIndex.value = props.zIndex ?? nextZIndex()
+    }
+  )
+
   watch(
     () => props.modelValue,
     (val) => {
       if (val) {
         closed.value = false
         open()
         rendered.value = true // enables lazy rendering
-        zIndex.value = isUndefined(props.zIndex) ? nextZIndex() : zIndex.value++
+        zIndex.value = props.zIndex ?? nextZIndex()
```

Comment:
The logic here doesn't need to be changed, or is your goal to stop the zIndex from increasing?

Reply:
I get it, that seems to make sense.